sympy/sets/tests/test_sets.py

```diff
@@ -366,6 +409,13 @@ def test_ProductSet_of_single_arg_is_arg():
     assert ProductSet(Interval(0, 1)) == Interval(0, 1)
 
 
+def test_ProductSet_is_empty():
+    assert ProductSet(S.Integers, S.Reals).is_empty == False
+    empty = Intersection(FiniteSet(sin(2)/pi), S.Integers)
+    assert empty is not S.EmptySet and \
+            ProductSet(empty, S.Reals).is_empty == None
```

Comment:
Rather than `is not S.EmptySet` should unchanged be used here?

Reply:
Well, I assume that I could put the following:
```
    assert unchanged(Intersection, FiniteSet(sin(2)/pi), S.Integers)
    empty = Intersection(FiniteSet(sin(2)/pi), S.Integers)
    assert ProductSet(empty, S.Reals).is_empty == None
```
but to me that seems to slightly obscure the intent of the test and it adds a separate assert that does not really fit in. Is there a better way to rewrite the test?

Reply:
I think just adding the assert unchanged line and keeping the rest would be fine. Using `is not` or `!=` in asserts is generally a very weak test of correctness (unless you are specifically testing `__eq__/__ne__`).

Looking closer though I'm not sure about this test. Ideally this would evaluate to the empty set and is_empty would give True. Perhaps a better example would be `empty = Intersection(FiniteSet(x), FiniteSet(2))` since if x is an arbitrary symbol then that necessarily has `is_empty == None`.

Reply:
Hmm, theoretically that'd be fine too, but I'm just noticing this (with current master):
```
>>> Intersection(FiniteSet(x), FiniteSet(2))
{2}
```
That's quite an ugly bug.
Ha! You've even filed it already yourself :-)
https://github.com/sympy/sympy/issues/16953

So I think for now the test is good enough? I've added the "unchanged" assert though.

Reply:
Okay so maybe that example isn't so good right now :)

How about  `Interval(x, 1)`?

Reply:
Ok, changed.